src/server/ua_services_monitoreditem.c

```diff
@@ -240,6 +243,45 @@ checkAdjustMonitoredItemParams(UA_Server *server, UA_Session *session,
     return UA_STATUSCODE_GOOD;
 }
 
+static UA_StatusCode
+checkEventFilterParam(UA_Server *server, UA_Session *session,
+                      const UA_MonitoredItem *mon,
+                      UA_MonitoringParameters *params){
+    if(mon->itemToMonitor.attributeId == UA_ATTRIBUTEID_EVENTNOTIFIER) {
+        UA_EventFilter *eventFilter =
+            (UA_EventFilter *) params->filter.content.decoded.data;
+        //check the where clause for logical consistency
+        if(eventFilter->whereClause.elementsSize != 0) {
+            UA_ContentFilterResult contentFilterResult;
+                UA_Event_staticWhereClauseValidation(server, &eventFilter->whereClause,
+                                                 &contentFilterResult);
+            for(size_t i = 0; i < contentFilterResult.elementResultsSize; ++i) {
+                if(contentFilterResult.elementResults[i].statusCode != UA_STATUSCODE_GOOD){
+                    //ToDo currently we return the first non good status code, check if
+                    //we can use the detailed contentFilterResult later
+                    UA_StatusCode whereResult =
+                        contentFilterResult.elementResults[i].statusCode;
+                    UA_ContentFilterResult_clear(&contentFilterResult);
+                    return whereResult;
+                }
+            }
+        }
+        //check the select clause for logical consistency
+        UA_STACKARRAY(UA_StatusCode, selectClauseValidationResult,
```

Comment:
This VLA can be abused to overload the stack.
Think a message with a million filter elements. That’s only a few MByte.
Either
- add a length limit (and use a fixed size temp array which can be more optimized by the compiler)
- Run the checks one-by-one and break upon the first problem. So no array at all.